src/explore-education-statistics-admin/src/pages/release/edit-release/content/helpers.ts

```diff
@@ -0,0 +1,345 @@
+import { KeyStatsFormValues } from '@admin/modules/editable-components/EditableKeyStatTile';
+import permissionsService from '@admin/services/permissions/service';
+import {
+  EditableContentBlock,
+  ExtendedComment,
+} from '@admin/services/publicationService';
+import { releaseContentService } from '@admin/services/release/edit-release/content/service';
+import {
+  ContentBlockAttachRequest,
+  ContentBlockPostModel,
+} from '@admin/services/release/edit-release/content/types';
+import { Dictionary } from '@admin/types';
+import {
+  AbstractRelease,
+  ContentSection,
+} from '@common/services/publicationService';
+import { AxiosResponse } from 'axios';
+import { RemoveBlockFromSection } from './actions';
+import { Dispatch } from './ReleaseContext';
+
+const contentSectionComments = (
+  contentSection?: ContentSection<EditableContentBlock>,
+) => {
+  if (
+    contentSection &&
+    contentSection.content &&
+    contentSection.content.length > 0
+  ) {
+    return contentSection.content.reduce<ExtendedComment[]>(
+      (allCommentsForSection, content) =>
```

Comment:
Use `acc` and push new `content`s into it. Creating new arrays on every iteration is not particularly efficient (a lot of code seems to be following this anti-pattern).